packages/api/cms-api/src/page-tree/createPageTreeResolver.ts

```diff
@@ -158,7 +158,10 @@ export function createPageTreeResolver({
                                 selection.typeCondition &&
                                 selection.typeCondition.kind == "NamedType"
                             ) {
-                                if (selection.typeCondition.name.value === node.documentType) {
+                                if (
+                                    selection.typeCondition.name.value === node.documentType ||
+                                    selection.typeCondition.name.value === DocumentInterface.name
```

Comment:
@nsams 
I'm not entirely sure if this change defeats the purpose of your optimization. Maybe it would be favourable to not query `DocumentInterface` but instead supply a more specific query using `Page` or `Link` to the `usePage` factory?